src/transformers/models/rag/modeling_tf_rag.py

```diff
@@ -1518,3 +1518,27 @@ def compute_loss(self, labels, y_pred, smooth_epsilon=0.0, from_logits=True, red
         loss = (1.0 - smooth_epsilon) * nll_loss + eps_i * smooth_loss
 
         return loss
+
+
+class TFWrappedModel:
+    """
+    this class wraps a the TFSharedEmbeddingTokens layer into a python 'no-keras-layer' class to avoid problem with
+    weight restoring. Also it makes sure that the layer is called from the correct scope to avoid problem with
+    saving/storing the correct weights
+    """
+
+    def __init__(self, layer, abs_scope_name=None):
+        self._layer = layer
+        self._abs_scope_name = abs_scope_name
+
+    def call(self, *args, **kwargs):
+        # if an abs scope name is given to the embedding variable, call variable from absolute scope
+        with tf.compat.v1.variable_scope(self._abs_scope_name, auxiliary_name_scope=False) as abs_scope_name:
+            with tf.name_scope(abs_scope_name.original_name_scope):
+                return self._layer.call(*args, **kwargs)
+
+    def __call__(self, *args, **kwargs):
+        # if an abs scope name is given to the embedding variable, call variable from absolute scope
+        with tf.compat.v1.variable_scope(self._abs_scope_name, auxiliary_name_scope=False) as abs_scope_name:
+            with tf.name_scope(abs_scope_name.original_name_scope):
+                return self._layer(*args, **kwargs)
```

Comment:
This is nowhere used, for what is this used for?

Reply:
I think Patrick already deleted this part in the latest version